scripts/lint_prowjobs/main.go

```diff
@@ -0,0 +1,290 @@
+	fmt.Println(gitDiffCommand)
+
+	gitDiffOutput, err := exec.Command("git", strings.Split(gitDiffCommand, " ")[1:]...).Output()
+	filesChanged := strings.Fields(string(gitDiffOutput))
+	for _, file := range filesChanged {
+		if strings.Contains(file, "presubmits") {
+			presubmitFiles = append(presubmitFiles, file)
+		}
+		if strings.Contains(file, "postsubmits") {
+			postsubmitFiles = append(postsubmitFiles, file)
+		}
+	}
+	return presubmitFiles, postsubmitFiles, err
+}
+
+func unmarshalJobFile(filePath string, jobConfig *config.JobConfig) (string, string, string, *config.JobConfig) {
+	jobOrgRepo := strings.Replace(filepath.Dir(filePath), "jobs/", "", 1)
+	jobFileName := filepath.Base(filePath)
+	fileContents, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Fatalf("Error reading contents of %s: %v", filePath, err)
+	}
+	fileContentsString := string(fileContents)
+	if err := yaml.Unmarshal(fileContents, &jobConfig); err != nil {
+		log.Fatalf("Error unmarshaling contents of %s: %v", filePath, err)
+	}
+	return jobOrgRepo, jobFileName, fileContentsString, jobConfig
+}
+
+func displayConfigErrors(fileErrorMap map[string][]string) bool {
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 0, '\t', 0)
+	errorsExist := false
+	for file, errors := range fileErrorMap {
+		if len(errors) > 0 {
+			errorsExist = true
+		}
+		fmt.Println()
+		fmt.Println(file + ":")
+		for _, err := range errors {
+			fmt.Fprintln(w, err)
+		}
+	}
+	w.Flush()
+	return errorsExist
+}
+
+func main() {
+	var jobConfig config.JobConfig
+	var presubmitErrors = make(map[string][]string)
+	var postsubmitErrors = make(map[string][]string)
+
+	presubmitConstants := new(JobConstants)
+	presubmitConstants.Init("presubmit")
+	postsubmitConstants := new(JobConstants)
+	postsubmitConstants.Init("postsubmit")
+
+	pullBaseSha := os.Getenv("PULL_BASE_SHA")
+	pullPullSha := os.Getenv("PULL_PULL_SHA")
+
+	gitRootOutput, _ := exec.Command("git", "rev-parse", "--show-toplevel").Output()
+	gitRoot := strings.Fields(string(gitRootOutput))[0]
+
+	presubmitFiles, postsubmitFiles, err := getFilesChanged(gitRoot, pullBaseSha, pullPullSha)
+	if err != nil {
+		log.Fatalf("There was an error running the git command: %v", err)
+	}
+
+	presubmitCheckFunctions := []presubmitCheck{
+		AlwaysRunCheck(),
+		PresubmitClusterCheck(presubmitConstants),
+		SkipReportCheck(),
+		PresubmitBucketCheck(presubmitConstants),
+		ServiceAccountCheck(presubmitConstants),
+		PresubmitMakeTargetCheck(presubmitConstants),
+	}
+
+	postsubmitCheckFunctions := []postsubmitCheck{
+		PostsubmitClusterCheck(postsubmitConstants),
+		PostsubmitBucketCheck(postsubmitConstants),
+		PostsubmitMakeTargetCheck(postsubmitConstants),
+	}
+
+	for _, presubmitFile := range presubmitFiles {
+		presubmitOrgRepo, presubmitFileName, fileContentsString, jobConfigMap := unmarshalJobFile(presubmitFile, &jobConfig)
+		presubmitJobs := jobConfigMap.PresubmitsStatic
+		presubmitConfig := presubmitJobs[presubmitOrgRepo][0]
```

Comment:
* nit: this can be onelined
* Be sure you're checking there is an value before referencing it, and that the list has members before referencing an index `[0]`
```suggestion
		presubmitJobs, ok :=  jobConfigMap.PresubmitsStatic[presubmitOrgRepo]
		if !ok {
		     // raise error that key isn't present
		}
		if len(presubmitJobs) < 1 {
		    // raise error or skip because empty
		}
```